### PR TITLE
#157944409 Build out unittests for requests.

### PR DESCRIPTION
**Pull request by joelmachango**
**What does this PR do?**
Developing the test cases for the request resource

**Description of Task to be completed**
1. Have an environment setup for creating unitttests
2. Have a suitable directory structure for my project.
3. Have requests unittests  for the following endpoints.

| Endpoint                                                | Functionality         | HTTP method |
| ------------------------------------------------------- | --------------------- | ----------- |
| http://[hostname]/requests/api/v1/requests              | Fetch all requests    | GET         |
| http://[hostname]/requests/api/v2/requests[request_id]  | Fetch a requests      | GET         |
| http://[hostname]/requests/api/v3/requests              | Create a requests     | POST        |
| http://[hostname]/requests/api/v4/requests[request_id]  | Modify a requests     | PUT         |

**How should this be manually tested?**
Clone the repository
`$ git clone https://github.com/joelmachango/maintenance-tracker`

`$git checkout ch-requests-unittests`

cd maintenance-tracker/app/

`$ python3 -m venv venv`

`$. venv/bin/activate`

`$ pip install -r requirements.txt`

**Any background context you want to provide?**
This PR consist of unittests for the request resource.

**What are the relevant pivotal tracker stories?**
[#157944409](https://www.pivotaltracker.com/n/projects/2173255/stories/157944409)

### CODE REVIEW

**Comment by lawrenceChege** [comment]
hey the commits are great.
how did you do the table? i like it

File: tests/test_requests.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    def test_create_request_successfully(self):
+        """tests if client can create (POST) a request"""        
+        route = self.client.post('http://[hostname]/requests/api/v3/requests[request_id]', data=dict (
+            request_id="001",
+            request="Bad internet",
+            request_type="repair"
+        ))
+        self.assertEqual(route.status_code, 201)
+
+    def test_modify_request_successfully(self):
```
Comment (daisymacharia) [inline tests/test_requests.py]: you are testing for a `PUT` request but the test only shows that you have the post after which you should now test the edit for the specific request.

**Comment by Slinjez** [comment]
Nice pr, super impressive

File: tests/test_requests.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        """tests if a user can modify (PUT) a request"""
+                route = self.client.post('http://[hostname]/requests/api/v4/requests[request_id]', data=dict (
```
Comment (daisymacharia) [inline tests/test_requests.py]: all the api urls you will be creating should have the same version ie ` http://[hostname]/requests/api/v1/requests[request_id]`, the versions change incase there is an update to your api then you can then move to the next version. But for this challenge, lets stick to using v1.

File: tests/test_requests.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    def setUp(self):
+        """creates a test client"""
+        self.app = app.test_client()
+        """propagate the exceptions to the test client"""
+        self.app.testing = True 
+
+    def test_fetch_all_requests_successfully(self):
+        """tests if client can GET all requests"""       
+        route = self.client.get('http://[hostname]/request/api/v1/requests')
++       self.assertEqual(route.status_code, 200)
```
Comment (daisymacharia) [inline tests/test_requests.py]: This is a good test :thumbs up:

File: tests/test_requests.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        
+    def test_fetch_a_request(self):
+        """tests if client can GET a particular request"""  
+        route = self.client.get('http://[hostname]/requests/api/v2/requests[request_id]')
```
Comment (daisymacharia) [inline tests/test_requests.py]: fix the api versioning to use v1

File: tests/test_requests.py (hunk 2 above)
Comment (joelmachango) [reply tests/test_requests.py]: Noted. Makes sense 👍 I'll edit this.

File: tests/test_requests.py (hunk 1 above)
Comment (joelmachango) [reply tests/test_requests.py]: Noted. I'll change this to` client.put`

**Pull request merged**